internal/provider/provider.go

```diff
@@ -373,6 +373,35 @@ func New() *schema.Provider {
 						Optional:      true,
 						ConflictsWith: []string{"command"},
 					},
+					"healthcheck": {
+						Type:          schema.TypeSet,
+						Description:   "HTTP health checking to determine the application readiness.",
+						ForceNew:      true,
+						Optional:      true,
+						ConflictsWith: []string{"command"},
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"url": {
+									Type:        schema.TypeString,
+									Description: "HTTP address used determine the application readiness. A successful health check is a HTTP response code less than 500 returned before healthcheck.interval seconds.",
+									ForceNew:    true,
+									Required:    true,
+								},
+								"interval": {
+									Type:        schema.TypeInt,
+									Description: "Duration in seconds to wait between healthcheck requests.",
+									ForceNew:    true,
+									Required:    true,
+								},
+								"threshold": {
+									Type:        schema.TypeInt,
+									Description: "Number of consecutive heathcheck failures before returning an unhealthy status.",
+									ForceNew:    true,
+									Required:    true,
+								},
```

Comment:
One minor nit before merge, should we make these optional instead with reasonable defaults?

Also Kubernetes calls this `period` instead of `interval`. I'm impartial, just wanted to toss the idea up.